22_pipeslab/pipeslab.txt

```diff
@@ -2,20 +2,47 @@ For each of the following, type a single command line that performs:
 
   1. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner.
   Answer: 
+I used ls -la to get all the content (including hidden)
+It works for me in such way:
+
+$ ls -la | cut -c 4 | grep x | cat -n | tail -1 | cut -f1
+
+or, using regex:
+
+$ ls -la | grep  '^...x' | cat -n | cut -f1 | tail -1
+
+But shorter is:
+
+$ ls -la | grep  '^...x' | wc -l
 
   2. Print the number of lines from "ls -l" which represent a path that has execute permission to their owner, group and others.
   Answer: 
+$ ls -la | cut -c 4,7,10 | grep xxx | wc -l
+
+or
+
+$ ls -la | grep  '^...x..x..x' | wc -l
 
   3. Print the number of lines from "ls -l" which represent paths that were modified today (no matter which day we run the command)
   Answer: 
+$ export LC_ALL=C
+$ ls -la | grep "$(date +"%b %d")"
 
   4. Modify (1) and (2) to count files only (no directories)
   Answer:
+1. $ ls -la -p | grep -v / | cut -c 4 | grep x | wc -l
+
+2. $ ls -la -p | grep -v / | cut -c 4,7,10 | grep xxx | wc -l
+
 
   5. The command /sbin/ifconfig shows information regarding the machine's network interfaces. Use the command to print only the IP address of the machine. Remember not to show the special loopback address 127.0.0.1
 
+$ /sbin/ifconfig | grep 'inet addr' | tail -1 | cut -f1 -d B | tr -d a-z | cut -f2 -d :
```

Comment:
I wouldn't remove the last one as 127.0.0.1 could be the first. Better to use `grep -v 127.0.0.1` to remove just that one